src/Quartz/Collection/TreeSet.cs

```diff
@@ -20,26 +20,37 @@
 using System;
 using System.Collections;
 using System.Collections.Generic;
+using System.Runtime.Serialization;
 
 namespace Quartz.Collection
 {
     /// <summary>
     /// Only for backwards compatibility with serialization!
     /// </summary>
     /// <typeparam name="T"></typeparam>
+#if BINARY_SERIALIZATION
     [Serializable]
+#endif // BINARY_SERIALIZATION
+    [DataContract]  // TODO (NetCore Port): Confirm that data contract serialization works as expected here
```

Comment:
We can probably remove many of these. They're here for two reasons - one of which is a good reason and one of which is not :)

The good reason: By default, Json.NET will serialize all public fields/properties (even if no DataContract attribute is present), but it won't serialize private members. If a DataContract attribute is applied, it will instead serialize all members attributed with DataMember (whether public or private) and nothing else. So, in some cases, using data contract annotation members is important to make sure we're serializing the same members that would have been serialized previously. I suspect that we have (many) more attributes than are really necessary just for that, though (this comment is on one such attribute which could be removed).

The not very good reason they're here: When I started working on this port, I was using DataContractSerialization instead of Json.Net and DCS requires many more data contract attributes. Because of that, I attributed pretty much everything. In the end, I switched to Json.Net because I found DCS is not very good at handling serialization of types that aren't known at compile time (if i included custom types in a JobDataMap it caused problems, for example). However, after switching to Json.Net the attributes were still left behind.

Now that we're not using DCS, many of these probably aren't necessary anymore. I've created mjrousos/quartznet#1 (and assigned it to myself) to track this cleanup